wnfs/src/private/share.rs

```diff
@@ -317,31 +317,38 @@ pub mod recipient {
     use wnfs_hamt::Hasher;
     use wnfs_namefilter::Namefilter;
 
-    /// Checks if a share count is available.
-    pub async fn find_share(
-        share_count: u64,
+    /// Seeks to the latest share counter that is populated.
+    pub async fn find_latest_share_counter(
+        share_count_start: u64,
         limit: u64,
         recipient_exchange_key: &[u8],
         sharer_root_did: &str,
         sharer_forest: &PrivateForest,
         store: &impl BlockStore,
     ) -> Result<Option<u64>> {
         for i in 0..limit {
-            let share_label = sharer::create_share_label(
-                share_count + i,
-                sharer_root_did,
-                recipient_exchange_key,
-            );
+            let share_count = share_count_start + i;
 
-            if sharer_forest
+            let share_label =
+                sharer::create_share_label(share_count, sharer_root_did, recipient_exchange_key);
+
+            if !sharer_forest
                 .has(&Sha3_256::hash(&share_label), store)
                 .await?
             {
-                return Ok(Some(share_count + i));
+                if i == 0 {
+                    // There don't seem to be any shares yet
+                    return Ok(None);
+                } else {
+                    // We've hit the first unpopulated label,
+                    // so the last one was the last valid share counter.
+                    return Ok(Some(share_count - 1));
+                }
```

Comment:
Nit: Cosmetic suggestion :)

```rs
return Ok(...)
```

Reply:
Hm, I don't understand?

Reply:
Are you referring to the last line? clippy actually complains if I put `return Ok(Some(...));` into the last line instead of `Ok(Some(...))`.

Reply:
Sorry. I mean 

```rs
return Ok(if i == 0 {
    // There don't seem to be any shares yet
    None
} else {
    // We've hit the first unpopulated label,
    // so the last one was the last valid share counter.
    Some(share_count - 1)
})
```

You can ignore if it doesn't appeal to your senses.

Reply:
Oh. I actually had it like this before, but ended up changing it.
Especially with the above change it's going to be a bigger check than `i == 0`, so perhaps more readable if it's not an expression :D
I'll keep it this way.

Reply:
That makes sense